client/src/pages/Train.jsx

```diff
@@ -294,7 +362,15 @@ class Train extends Component {
   switchComponent() {
     switch (this.state.gameState) {
       case Train.STATE_INTRO:
-        return <IntroductionView onClick={() => this.getNewQuestion()} />;
+        return (
+          <IntroductionView
+            onClick={(system, difficulty) => {
+              this.setState({ system: system });
+              this.setState({ difficulty: difficulty });
+              this.getNewQuestion();
+            }}
```

Comment:
Ici on a des soucis car `setState` est asynchrone, donc le state n'est pas forcément modifié avant de générer la question, c'est pour ça qu'on a des soucis avec la première question.
On peut essayer ça 
```suggestion
            onClick={(system, difficulty) => {
              this.setState({ system: system, difficulty: difficulty }, () => this.getNewQuestion());
            }}
```

Reply:
Normalement `setState` prend une callback en second argument, je sais pas si ça marche comme ça, essaie sinon j'ai une autre solution un peu plus chiante